src/wfqc/metrics.py

```diff
@@ -1,81 +1,313 @@
 import igraph
+import numpy as np
 
-def sum_metric_citation(graph, workflow): # cocitation metric 
 
-    """
-    Calculates the cocitation  metric for a given workflow and a given citation (OBS not cocitation) graph
-    
-    Parameters
-    ----------
-    graph : igraph.Graph
-        Graph generated by igraph
-    workflow, list of tuples with pairs of strings TODO: how do I write this? does it matter I wont have this format later anyways? 
-        List of tuples of strings corresponding to the edges in the workflow. 
 
-    """
-    # List to  collect pairwise scores
-    score_list = [] # TODO: can predefine the list length, does not matter this is temporary?
+def get_node_ids(graph: igraph.Graph, key= "name"):
```

Comment:
It is a good practice to use typing to describe the data types, apart from better documentation and readability of the function it is possible to use type checking libraries such as `pytype` to ensure that the code uses the data types as they are intended, e.g., a `string` is never expected to be an `int`, or a `list` is never expected to be a `dict`. In addition, the methods should include docstrings, describing the method, each of the parameters, and the return value.

```suggestion
def get_node_ids(graph: igraph.Graph, key:str= "name") -> dict:
```